metamath.tex

```diff
@@ -7440,31 +7439,42 @@ \subsection{Preprocessing}
 \texttt{\$(} begins a {\bf comment} and \texttt{\$)} ends a comment.\index{\texttt{\$(}
 and \texttt{\$)} auxiliary keywords}\index{comment}  Comments may contain any of
 the 94 printable characters and white space, except they may not contain the
-2-character sequences \texttt{\$(} or \texttt{\$)}.  Comments are ignored (treated
-like white space) for the purpose of parsing.
-
-Two special characters inside comments, \verb$`$ and \verb$~$, control the
-typesetting of comments and are discussed on p.~\pageref{mathcomments}. They
-may be ignored for the purpose of parsing.
+2-character sequences \texttt{\$(} or \texttt{\$)} (comments do not nest).  Comments are ignored (treated
+like white space) for the purpose of parsing, e.g.,
+\texttt{\$( \$[ \$)} is a comment.
+See p.~\pageref{mathcomments} for comment typesetting conventions; these
+conventions may be ignored for the purpose of parsing.
 
 A {\bf file inclusion command} consists of \texttt{\$[} followed by a file name
 followed by \texttt{\$]}.\index{\texttt{\$[} and \texttt{\$]} auxiliary
-keywords}\index{included file}\index{file inclusion}  The file name may not
-contain a \texttt{\$} or white space.  The file must exist.  The case-sensitivity
+keywords}\index{included file}\index{file inclusion}
+It is only allowed in the outermost scope (i.e., not between 
+\texttt{\$\char`\{} and \texttt{\$\char`\}})
+and must not be inside of a statement (e.g., it may not occur
+between the label of a \texttt{\$a} statement and its \texttt{\$.}).
+The file name may not
+contain a \texttt{\$} or white space.  The file must exist, and is
+relative to the current directory (which might not be the directory
```

Comment:
The problem of directory paths is discussed in the thread of this post:
https://groups.google.com/d/msg/metamath/eI0PE0nPOm0/f8_btXt4AgAJ
In Stefan's post prior to this, he suggests we can "ust use underscores to separate semantic levels" which is what we are doing now.  I'm not sure how to word it, but I think we should discourage directory paths in file names at this point (if not forever).